actors/market/src/lib.rs

```diff
@@ -248,11 +249,15 @@ impl Actor {
         let state: State = rt.state()?;
 
         for (di, mut deal) in params.deals.into_iter().enumerate() {
-            // drop malformed deals
-            if let Err(e) = validate_deal(rt, &deal, &network_raw_power, &baseline_power) {
-                info!("invalid deal {}: {}", di, e);
-                continue;
-            }
+            let serialized_proposal =
+                match validate_deal(rt, &deal, &network_raw_power, &baseline_power) {
+                    // drop malformed deals
+                    Err(e) => {
+                        info!("invalid deal {}: {}", di, e);
+                        continue;
+                    }
+                    Ok(b) => b,
```

Comment:
Ok a design question here is whether the deal proposal sent to the notification should be normalized, or the raw thing the client sent. I think it should be normalised. This happens at line 311, but could be brought up a lot earlier.

The `validate_deal` function needs to work on the raw proposal (just for the authentication part), but everything else could use normalized, so there's some freedom in sequencing. 

I can see you might be trying to avoid serializing the normalized one but worry not! We already do at line 313 to calculate its CID for duplicate detection (which accounts for address aliases).

Reply:
aha nice, I was indeed trying to not add an expensive extra serialization